File: crabzilla-core/src/main/java/crabzilla/model/EntityId.java
package crabzilla.model;

import com.fasterxml.jackson.annotation.JsonTypeInfo;

import java.io.Serializable;

@JsonTypeInfo(use=JsonTypeInfo.Id.CLASS, include=JsonTypeInfo.As.PROPERTY, property="@class")
public interface EntityId extends Serializable {
  String stringValue();
}

File: crabzilla-core/src/main/java/io/github/crabzilla/core/entity/EntityId.kt
package io.github.crabzilla.core.entity

import com.fasterxml.jackson.annotation.JsonTypeInfo

import java.io.Serializable

@JsonTypeInfo(use = JsonTypeInfo.Id.CLASS, include = JsonTypeInfo.As.PROPERTY, property = "@class")
interface EntityId : Serializable {
  fun stringValue(): String
}
